messaging/kafka/client/syncproducer.go

```diff
@@ -124,10 +118,12 @@ func (ref *SyncProducer) Close() error {
 	}
 	ref.Debug("SyncProducer closed")
 
-	err = ref.Client.Close()
-	if err != nil {
-		ref.Errorf("client close error: %v", err)
-		return err
+	if !ref.Client.Closed() {
```

Comment:
I don't think it's a good idea to close always close the client. It might still be used somewhere else.
I would suggest to close the client only if it was create in the `New*Producer` function.
This applies for both producer types.